src/main/java/de/konfidas/ttc/messages/LogMessage.java

```diff
@@ -143,8 +144,14 @@ private int getEncodedLength(ASN1Primitive element) throws IOException, ExtendLe
             if (elementNumberOfLengthBytes > 4) {
                 throw new ExtendLengthValueExceedsInteger("Der Wert der extended length überschreitet einen Integer", null);
             }
-            byte[] lengthBytes = Arrays.copyOfRange(elementContent, 1, elementNumberOfLengthBytes + 1);
-            return ByteBuffer.wrap(lengthBytes).getInt();
+
+            byte[] lengthBytesFromElement = Arrays.copyOfRange(elementContent, 2, 2+elementNumberOfLengthBytes); //we need to have 4 bytes for an integer
+            byte[] prependBytes = new byte[4-elementNumberOfLengthBytes];
+
+            ByteBuffer lengthByte = ByteBuffer.wrap(new byte[4]);
+            lengthByte.put(prependBytes);
+            lengthByte.put(lengthBytesFromElement);
+            return lengthByte.getInt(0);
```

Comment:
why not

`return new BigInteger(lengthBytesFromElement).intValue();`

instead of this?

Reply:
Das geht leider nicht :-( Das Problem ist, dass die Kodierung der Länge Little-Endian ist. BigInteger unter Java aber Big-endian. Diese Lösung oben war das Einfachste, was mir einfiel.

Reply:
The Endcoding is:

"Long form. Two to 127 octets. Bit 8 of first octet has value "1" and bits 7-1 give the number of additional length octets. Second and following octets give the length, base 256, most significant digit first."
(cf. Section 3.1 of  http://luca.ntop.org/Teaching/Appunti/asn1.html), i.e. BigEndian. 

Anyhow, a length of `[0x82, 0x01, 0x02]` should be interpreted as `1* 256 + 2*1 = 258`, right?
And  `System.out.println(new BigInteger(new byte[]{0x01, 0x02}));` yields 258... which is, by the way, the same as
` System.out.println(ByteBuffer.wrap(new byte[]{0x00, 0x00, 0x01, 0x02}).getInt());`